app/models/course/leaderboard_settings.rb

```diff
@@ -35,13 +38,45 @@ def display_user_count
     @settings.display_user_count || 30
   end
 
-  # Returns the number of users to be displayed on the leaderboard
+  # Set the number of users to be displayed on the leaderboard
   #
   # @param [Integer] count The number of users to be displayed
   def display_user_count=(count)
     @settings.display_user_count = count
   end
 
+  # Returns whether group leaderboard is enabled (disabled by default).
+  #
+  # @return [Boolean] Setting on whether group leaderboard is enabled.
+  def enable_group_leaderboard
+    @group_leaderboard_settings.settings(:display) == true
```

Comment:
To be consistent with other settings, also use `:enabled` here ? 

And does `@group_leaderboard_settings.enabled == true` work ?